### PR TITLE
Add TypeTranslator to Hive connector

### PR DESCRIPTION
**Pull request by geraint0923**
The specific Hive connector could change their type-to-HiveType
conversion rules by implementing a new TypeInfoResolver.

### CODE REVIEW

**Comment by cberner** [comment]
Looks good

File: presto-hive/src/main/java/com/facebook/presto/hive/TypeInfoResolver.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.spi.type.Type;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+
+public interface TypeInfoResolver
```
Comment (martint) [inline presto-hive/src/main/java/com/facebook/presto/hive/TypeInfoResolver.java]: BTW, I'd call this interface "TypeTranslator"

File: presto-hive/src/main/java/com/facebook/presto/hive/TypeInfoResolver.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+{
+    TypeInfo resolveType(Type type);
```
Comment (martint) [inline presto-hive/src/main/java/com/facebook/presto/hive/TypeInfoResolver.java]: and this method "translate"

**Comment by geraint0923** [comment]
Renamed to `TypeTranslator` and `translate`.

**Pull request merged**